pkg/util/loadbalancer/loadbalancer.go

```diff
@@ -28,3 +37,124 @@ func RemoveFrontendIPConfiguration(lb *mgmtnetwork.LoadBalancer, resourceID stri
 func isFrontendIPConfigReferenced(fipConfig mgmtnetwork.FrontendIPConfiguration) bool {
 	return fipConfig.LoadBalancingRules != nil || fipConfig.InboundNatPools != nil || fipConfig.InboundNatRules != nil || fipConfig.OutboundRules != nil
 }
+
+// Remove outbound-rule-v4 IPs and corresponding frontendIPConfig from load balancer
+func RemoveOutboundIPsFromLB(lb mgmtnetwork.LoadBalancer) {
+	removeOutboundRuleV4FrontendIPConfig(lb)
+	setOutboundRuleV4(lb, []mgmtnetwork.SubResource{})
+}
+
+func removeOutboundRuleV4FrontendIPConfig(lb mgmtnetwork.LoadBalancer) {
+	var savedFIPConfig = make([]mgmtnetwork.FrontendIPConfiguration, 0, len(*lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations))
+	var outboundRuleFrontendConfig = getOutboundRuleV4FIPConfigs(lb)
+
+	for i := 0; i < len(*lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations); i++ {
+		fipConfigID := *(*lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations)[i].ID
+		fipConfig := (*lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations)[i]
+		hasLBRules := (*lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations)[i].LoadBalancingRules != nil
+		if _, ok := outboundRuleFrontendConfig[fipConfigID]; ok && !hasLBRules {
+			continue
+		}
+		savedFIPConfig = append(savedFIPConfig, fipConfig)
+	}
+	lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations = &savedFIPConfig
+}
+
+func getOutboundRuleV4FIPConfigs(lb mgmtnetwork.LoadBalancer) map[string]mgmtnetwork.SubResource {
+	var obRuleV4FIPConfigs = make(map[string]mgmtnetwork.SubResource)
+	for _, obRule := range *lb.LoadBalancerPropertiesFormat.OutboundRules {
+		if *obRule.Name == OutboundRuleV4 {
+			for i := 0; i < len(*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations); i++ {
+				fipConfigID := *(*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations)[i].ID
+				fipConfig := (*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations)[i]
+				obRuleV4FIPConfigs[fipConfigID] = fipConfig
+			}
+			break
```

Comment:
There's a lot of refactors at once, so I want to be sure we're not breaking things...

In the original code, /pkg/cluster/loadbalancerprofile.go L110, we had:

```
            // clear outbound-rule-v4 frontend ip config
			*obRule.FrontendIPConfigurations = []mgmtnetwork.SubResource{}
            break
```

why was this removed here?

Reply:
In multiple locations throughout this file, any time I needed to update the outbound rule frontendIPConfiguration I was iterating on outbound rules, matching on "outbound-rule-v4", and then setting the frontendIPConfiguration.  I created a setter function to do this now 
https://github.com/Azure/ARO-RP/pull/3157/files#diff-d87786de7031b1f89a1bce960b3176383b16135b6d8cf887898443fc348f2c21R173